test/e2e/storage/csistoragecapacity.go

```diff
@@ -0,0 +1,209 @@
+					Labels: map[string]string{
+						"test": f.UniqueName,
+					},
+				},
+				StorageClassName: scName,
+			}
+		}
+		csc := newCSIStorageCapacity("-csc1")
+		csc2 := newCSIStorageCapacity("-csc2")
+		csc3 := newCSIStorageCapacity("-csc3")
+
+		// Discovery
+
+		ginkgo.By("getting /apis")
+		{
+			discoveryGroups, err := f.ClientSet.Discovery().ServerGroups()
+			framework.ExpectNoError(err)
+			found := false
+			for _, group := range discoveryGroups.Groups {
+				if group.Name == storagev1.GroupName {
+					for _, version := range group.Versions {
+						if version.Version == cscVersion {
+							found = true
+							break
+						}
+					}
+				}
+			}
+			framework.ExpectEqual(found, true, fmt.Sprintf("expected CSIStorageCapacity API group/version, got %#v", discoveryGroups.Groups))
+		}
+
+		ginkgo.By("getting /apis/storage.k8s.io")
+		{
+			group := &metav1.APIGroup{}
+			err := f.ClientSet.Discovery().RESTClient().Get().AbsPath("/apis/storage.k8s.io").Do(context.TODO()).Into(group)
+			framework.ExpectNoError(err)
+			found := false
+			for _, version := range group.Versions {
+				if version.Version == cscVersion {
+					found = true
+					break
+				}
+			}
+			framework.ExpectEqual(found, true, fmt.Sprintf("expected CSIStorageCapacity API version, got %#v", group.Versions))
+		}
+
+		ginkgo.By("getting /apis/storage.k8s.io/" + cscVersion)
+		{
+			resources, err := f.ClientSet.Discovery().ServerResourcesForGroupVersion(storagev1.SchemeGroupVersion.String())
+			framework.ExpectNoError(err)
+			found := false
+			for _, resource := range resources.APIResources {
+				switch resource.Name {
+				case "csistoragecapacities":
+					found = true
+				}
+			}
+			framework.ExpectEqual(found, true, fmt.Sprintf("expected csistoragecapacities, got %#v", resources.APIResources))
+		}
+
+		// Main resource create/read/update/watch operations
+
+		ginkgo.By("creating")
+		createdCSC, err := cscClient.Create(context.TODO(), csc, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		_, err = cscClient.Create(context.TODO(), csc, metav1.CreateOptions{})
+		framework.ExpectEqual(apierrors.IsAlreadyExists(err), true, fmt.Sprintf("expected 409, got %#v", err))
+		_, err = cscClient.Create(context.TODO(), csc2, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("watching")
+		framework.Logf("starting watch")
+		cscWatch, err := cscClient.Watch(context.TODO(), metav1.ListOptions{LabelSelector: "test=" + f.UniqueName})
+		framework.ExpectNoError(err)
+
+		// added for a watch
+		_, err = cscClient.Create(context.TODO(), csc3, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("getting")
+		gottenCSC, err := cscClient.Get(context.TODO(), csc.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(gottenCSC.UID, createdCSC.UID)
+
+		ginkgo.By("listing")
+		cscs, err := cscClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "test=" + f.UniqueName})
```

Comment:
should also list/watch across namespaces to exercise those two endpoints

Reply:
Done.

I've force-pushed to keep this in a mergeable state once it has LGTM, here are the changes: https://github.com/kubernetes/kubernetes/compare/65aee2038850a980500f1ccb25cf5dfc781748e7..ff47fb98735caf3578b7720cc806d2d2c195d684